pyproj/_crs.pyx

```diff
@@ -2883,9 +2879,14 @@ cdef class _CRS(Base):
                             out_confidence_list[iii]
                         )
                     )
+                # at this point, the auth name is copied and we can release the proj object
+                proj_destroy(proj)
+                proj = NULL
         finally:
-            for iii in range(num_proj_objects):
```

Comment:
This is here because the entire list may not be iterated through and not all of the objects will be accessed due to the confidence filter.

Reply:
As I understand the proj docs, every time you call `proj_list_get` you will get a copy of the element that you need to clean up.

Reply:
Yeah, see https://github.com/OSGeo/PROJ/blob/master/src/iso19111/c_api.cpp#L8313

Reply:
Yes, you are correct. I see it calls `pj_obj_create` on each call. So, you only need to delete on access.